sql/core/src/main/scala/org/apache/spark/sql/execution/dynamicpruning/PartitionPruning.scala

```diff
@@ -48,6 +49,10 @@ import org.apache.spark.sql.execution.datasources.{HadoopFsRelation, LogicalRela
  */
 object PartitionPruning extends Rule[LogicalPlan] with PredicateHelper with JoinSelectionHelper {
 
+  private val buildBroadcastThreshold = math.max(
+    AUTO_BROADCASTJOIN_THRESHOLD.defaultValue.getOrElse(conf.autoBroadcastJoinThreshold),
```

Comment:
Why it must be smaller than 10 mb even if users have changed the value of `autoBroadcastJoinThreshold`? I don't understand the rationale here.

Reply:
To avoid disable DPP by setting `autoBroadcastJoinThreshold ` to a small value. For example:
1. If user set autoBroadcastJoinThreshold to -1 and has benefit:
   Before this PR:
   spark.sql.optimizer.dynamicPartitionPruning.reuseBroadcastOnly | filtering side data size <= 10MB | filtering side data size > 10MB
   -- | -- | --
   true | N | N
   false | Y | Y

   After this PR:
   spark.sql.optimizer.dynamicPartitionPruning.reuseBroadcastOnly | filtering side data size <= 10MB | filtering side data size > 10MB
   -- | -- | --
   true | N | N
   false | Y | N


2. If user set autoBroadcastJoinThreshold to 100MB and has benefit:
   Before this PR:
   spark.sql.optimizer.dynamicPartitionPruning.reuseBroadcastOnly | filtering side data size <= 100MB | filtering side data size > 100MB
   -- | -- | --
   true | Y | N
   false | Y | Y

   After this PR:
   spark.sql.optimizer.dynamicPartitionPruning.reuseBroadcastOnly | filtering side data size <= 100MB | filtering side data size > 100MB
   -- | -- | --
   true | Y | N
   false | Y | N

`Y` means will insert a DPP, `N` means will not insert a DPP. It is much reasonable than before.

Reply:
> To avoid disable DPP by setting autoBroadcastJoinThreshold to a small value. 

Do we really need to consider that? In practice, no one does it AFAIK. It's hard to define `canBroadcast` when `autoBroadcastJoinThreshold` is -1. I think it's more reasonable to simple use `conf.autoBroadcastJoinThreshold`

Reply:
OK, changed it to `(canBroadcastBySize(right, conf) || hintToBroadcastRight(hint))`.